libraries/iterate/src/iterate.js

```diff
@@ -218,18 +222,10 @@ class Iterate extends events.EventEmitter {
     });
     this.keepGoing = true;
 
-    // Two reasons we call it this way:
-    //   1. first call should have same exec env as following
-    //   2. start should return immediately
-    this.currentTimeout = setTimeout(async () => {
-      debug('starting iteration');
-      this.emit('started');
-      try {
-        await this.iterate();
-      } catch (err) {
-        console.error(err.stack || err);
-      }
-    }, 0);
+    return new Promise(resolve => {
+      this.once('started', resolve);
+      process.nextTick(() => this.iterate());
```

Comment:
Maybe a comment to explain the `nextTick` here?

Reply:
I'm not sure why.. I'll remove it :)